Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/type/rapidantigen/RACoronaTest.kt

```diff
@@ -51,20 +52,30 @@ data class RACoronaTest(
     @SerializedName("dateOfBirth")
     val dateOfBirth: LocalDate?,
 
+    @SerializedName("outdatedAfter")
+    val outdatedAfter: Duration,
```

Comment:
A test is outdated after `testedAt` + `TimeFromConfigConfigTime`.
`testedAt` is fixed, when the test was scanned.
`TimeFromConfigConfigTime` is dynamic and can change whenever the config changes.

I think it would be better to not  store a fixed value `outdatedAfter`, but have a function to call with the current arguments.

e.g. `fun getState(nowUTC, currentConfig): State`

edit:
Thinking about it, the `testedAt` is from an external time source. When determining the `OUTDATED`, we should probably also check `currentConfigisDeviceTimeCorrect`? /cc @mlenkeit

Reply:
sure, i was not sure which way makes more sense, should a test that was made when 48 hours were the config expire on the next day if the config is changed to be 12 hours?

checking device time probably makes sense, how would we handle a mismatch?

Reply:
I agree with Matthias, dynamic evaluation would make sense here.

Device time check is currently not expected for this I think but let me double check...

Reply:
sounds good, removed the outdatedAfter property and switched check to use appconfig
ill keep this unresolved untill there is info on the time check @mlenkeit

Reply:
No need to check the device time.